web/src/typing_data.ts

```diff
@@ -3,63 +3,64 @@ import * as util from "./util";
 
 // See docs/subsystems/typing-indicators.md for details on typing indicators.
 
-const typist_dct = new Map();
-const inbound_timer_dict = new Map();
+const typist_dct: Map<string, number[]> = new Map();
+// eslint-disable-next-line no-undef
+const inbound_timer_dict: Map<string, NodeJS.Timeout | undefined> = new Map();
 
-export function clear_for_testing() {
+export function clear_for_testing(): void {
     typist_dct.clear();
     inbound_timer_dict.clear();
 }
 
-function to_int(s) {
+function to_int(s: string): number {
     return Number.parseInt(s, 10);
 }
 
-function get_key(group) {
+function get_key(group: string[]): string {
     const ids = util.sorted_ids(group);
     return ids.join(",");
 }
 
-export function add_typist(group, typist) {
+export function add_typist(group: string[], typist: string): void {
```

Comment:
I’m pretty sure we actually call these with `typist: number`, and the `to_int` is overdefensive nonsense that should be removed.

Reply:
@andersk I looked at tests for `typing_data` and found this comment there:
```
test("basics", () => {
    // The typing_data needs to be robust with lists of
    // user ids being in arbitrary sorting order and
    // possibly in string form instead of integer. So all
    // the apparent randomness in these tests has a purpose.
```

So if we remove `to_int` calls we would have to change some tests is that ok?

Reply:
Yeah, while the sort order part is accurate, I think the "string form" bit is probably defensive programming that is not needed in TypeScript.

Can you audit the code paths calling into this module, just in case?

Reply:
Yeah sure checking it!

Reply:
@andersk @timabbott Yup on checking the use cases we are dealing with "number" type everywhere in `typing_data` module! So I will remove `to_int`, also I have one more small doubt should I also change `util.sorted_ids` to accept only `number[]` instead of `(string | number)[]` there isn't any other module which is passing "string" ids to this function right?